### ISSUE TITLE
CS: Condition Zero fails to start after install

### ISSUE THREAD

**Issue by fjeddy** [GameServerManagers/LinuxGSM#1450]
Attempted roughly 50 different re-installs the last 72 hours on both Ubuntu 16.04, Ubuntu 14.04 and Ubuntu 12.04 using both 64-bit and 32-bit. As well as Debian 7.

All of them, regardless of how many validates and re-installs i attempt, return the following on debug

```
Console initialized.
Using breakpad crash handler
Setting breakpad minidump AppID = 80
Forcing breakpad minidump interfaces to load
Looking up breakpad interfaces from steamclient
Calling BreakpadMiniDumpSystemInit
FATAL ERROR (shutting down): W_LoadWadFile: couldn't load gfx.wad
email debug.log to linux@valvesoftware.com
Sat May 27 20:01:49 UTC 2017: Server restart in 10 seconds
```

Installation of a Condition Zero server using LinuxGSM can therefor be considered broken.

**Comment by UltimateByte** [comment]
Reproduced the issue on Debian 8 x64.
Not sure where it comes from, maybe a change from Valve.
https://steamdb.info/app/90/history/

**Comment by fjeddy** [comment]
The thread you are referring to is from 2014, this problem is much more recent. Unless of course the same has happend again.

**Comment by UltimateByte** [comment]
The thread i linked is also for windows. But since i got no clue, i consider any possibility.

**Comment by UltimateByte** [comment]
Same issue with 1.6 it seems.
#1457 1457

**Comment by Twonkk** [comment]
Thanks @ultimatebyte, wasn't sure if I should have posted here with the CZ issue or just make my own...

Anyways, I did some testing today and did manage to make a work around.. I am in the process of replicating my workaround.. if I can do it again, I'll post and try and get a pull request if I can..

**Comment by UltimateByte** [comment]
@Twonkk CZ and 1.6 are pretty similar in the way they're handled by SteamCMD and of course by LGSM.

Good to hear that you found a way to get it to work. Please share when you find it back.

**Comment by Twonkk** [comment]
@UltimateByte, thanks to @Gamma_Gandalph in Discord, using steam account with steam cmd fix's both games.. 

steps i used are;
>log into user account that manages CS or CZ
>rm -R serverfiles <--- Just to be safe
>cd steamcmd
>./steamcmd.sh
>login user password <-- MUST USE STEAM ACCOUNT WITH ACCESS TO GAME
>force_install_dir ../serverfiles <--- Thats two periods and forward slash  
>app_update 90 validate
>quit
>cd ../ <--- again, two periods and forward slash

>./csserver start 

works no problem now.. Looks like CS/CZ scripts need an update for steam accounts. i don't even know where to start to put a pull request in for... =D

**Comment by broonu** [comment]
This workaround didn't work here for CS 1.6.

```
csserver@mediaserver:~/steamcmd$ ./steamcmd.sh
Redirecting stderr to '/home/csserver/Steam/logs/stderr.txt'
[  0%] Checking for available updates...
[----] Verifying installation...
Steam Console Client (c) Valve Corporation
-- type 'quit' to exit --
Loading Steam API...OK.

Steam>login broonu ***************
Logging in user 'broonu' to Steam Public...
Enter the current code from your Steam Guard Mobile Authenticator app
Two-factor code:*****
Logged in OK
Waiting for license info...OK

Steam>force_install_dir ../serverfiles

Steam>app_update 90 validate
 Update state (0x5) validating, progress: 0.00 (0 / 66393415)
 Update state (0x5) validating, progress: 75.86 (50364437 / 66393415) <---- this never gets 100%
Success! App '90' fully installed.

Steam>quit

Steam>csserver@mediaserver:~/steamcmd$ cd ..
csserver@mediaserver:~$ ./csserver d

Counter-Strike 1.6 Debug
=================================

Distro: Debian GNU/Linux 8 (jessie)
Arch:   x86_64
Kernel: 3.16.0-4-amd64
Hostname:       mediaserver
tmux:   tmux 1.9
Avg Load:       0.30, 0.38, 0.34
Free Memory:    4.6G
Free Disk:      9.4G
GLIBC required: 2.3.4
Server IP:      AAA.BBB.CCC.DDD:27015
Server password:        NOT SET

Start parameters:
./hlds_run -game cstrike -strictportbind +ip AAA.BBB.CCC.DDD -port 27015 +clientport 27005 +map de_dust2 -maxplayers 16 -debug

Use for identifying server issues only!
Press CTRL+c to drop out of debug mode.
Warning! If cs-server is already running it will be stopped.

Continue? [Y/n] Y
[ INFO ] Debug cs-server: Stopping any running servers
[ INFO ] Stopping cs-server: LinuxGSM is already stopped
[  OK  ] Debug cs-server: Starting debug
Enabling debug mode
Auto-restarting the server on crash

Console initialized.
Using breakpad crash handler
Setting breakpad minidump AppID = 10
Forcing breakpad minidump interfaces to load
Looking up breakpad interfaces from steamclient
Calling BreakpadMiniDumpSystemInit
FATAL ERROR (shutting down): W_LoadWadFile: couldn't load gfx.wad
email debug.log to linux@valvesoftware.com
Mon Jun  5 16:37:31 BRT 2017: Server restart in 10 seconds

```

**Comment by Twonkk** [comment]
But did you >rm -R serverfiles/   ???

**Comment by broonu** [comment]
Yes, I did.
They closed my post about this issue, but it's not working even using steam account.
I installed Steam in my windows machine, downloaded the game and uploaded the folder to linux server and it's working.

Update state (0x5) validating, progress: 75.86 (50364437 / 66393415) <---- this never gets 100%
With LinuxGSM the download is always incomplete.

**Comment by UltimateByte** [comment]
@broonu We closed because your post was a duplicate of this one.

**Comment by MarkL4YG** [comment]
@broonu I would call SteamCMD not completing its validation phase a bug of LGSM...

**Comment by josecarlosweb** [comment]
I have the same problem. I have ubuntu x64 16.04LTS. Please, someone help me?

**Comment by MarkL4YG** [comment]
> email debug.log to linux@valvesoftware.com
So... what does that log say?

**Comment by Twonkk** [comment]
@josecarlosweb i can tell you, it works if it's done correctly. i have used that method multiple times.. Debian, Ubuntu, CentOS servers...

**Comment by josecarlosweb** [comment]
Sorry, i'm trying install on droplet of digital ocean. The same problem is here. The error is:
FATAL ERROR (shutting down): W_LoadWadFile: couldn't load gfx.wad.
The out of debug is:

```
Console initialized.
Using breakpad crash handler
Setting breakpad minidump AppID = 80
Forcing breakpad minidump interfaces to load
Looking up breakpad interfaces from steamclient
Calling BreakpadMiniDumpSystemInit
FATAL ERROR (shutting down): W_LoadWadFile: couldn't load gfx.wad
email debug.log to linux@valvesoftware.com
Sat Jun 17 15:52:32 UTC 2017: Server restart in 10 seconds

```
The download never complete, no matter how many times I've tried.

**Comment by Twonkk** [comment]
if you follow the post above i made, and add this into it..

app_update 90 +app_set_config 90 mod czero validate

**Comment by josecarlosweb** [comment]
The out is:
```
czserver@czserver:~/steamcmd$ rm -R ../serverfiles
czserver@czserver:~/steamcmd$ ./steamcmd.sh +login touchmx ****** +force_install_dir /home/czserver/serverfiles +app_update 90 +app_set_config 90 mod czero validate +quit

Redirecting stderr to '/home/czserver/Steam/logs/stderr.txt'
[  0%] Checking for available updates...
[----] Verifying installation...
Steam Console Client (c) Valve Corporation
-- type 'quit' to exit --
Loading Steam API...OK.

Logging in user 'touchmx' to Steam Public...Logged in OK
Waiting for user info...OK
 Update state (0x3) reconfiguring, progress: 0.00 (0 / 0)
 Update state (0x61) downloading, progress: 1.44 (954176 / 66393415)
 Update state (0x61) downloading, progress: 47.76 (31711578 / 66393415)
Success! App '90' fully installed.
czserver@czserver:~/steamcmd$
```

**Comment by Twonkk** [comment]
What's the output for ./csczserver debug ?

**Comment by Moshifan100** [comment]
The output is the same as above, which you might not have seen, so here it is

`Console initialized.
Using breakpad crash handler
Setting breakpad minidump AppID = 80
Forcing breakpad minidump interfaces to load
Looking up breakpad interfaces from steamclient
Calling BreakpadMiniDumpSystemInit
FATAL ERROR (shutting down): W_LoadWadFile: couldn't load gfx.wad
email debug.log to linux@valvesoftware.com
Sat Jun 17 15:52:32 UTC 2017: Server restart in 10 seconds
`

I get this same error on DigitalOcean.

**Comment by Twonkk** [comment]
@moshifan100

I had no issues with the csczserver file but others seam too be, so a solution to your issue would be to follow this

>log into user account that manages CS or CZ
rm -R serverfiles <--- Just to be safe
cd steamcmd
./steamcmd.sh
login user password <-- MUST USE STEAM ACCOUNT WITH ACCESS TO GAME
force_install_dir ../serverfiles <--- Thats two periods and forward slash
app_update 90 +app_set_config 90 mod czero validate
quit
cd ../ <--- again, two periods and forward slash

>./csczserver start

**Comment by Moshifan100** [comment]
I tried this, it failed. When I tried to start it, there were missing files and this only happened when I did that. This did not happen when I used anonymous account.

**Comment by fjeddy** [comment]
@Moshifan100 Are you looking to fix this or just looking to get your server up and running? I got it running by just fetching a backup from earlier and set it up as a new server, can provide you with a link to download the tar file if your sole goal is to get a server online you can play on.

**Comment by Moshifan100** [comment]
I'm just looking to set up a server, so it would be great if you could send me the tar file.

**Comment by broonu** [comment]
You could copy the folder from a working game in windows to your server. It
did the trick here, cause the install never work here, never gets 100% game
download.

2017-06-21 5:41 GMT-03:00 Alexander Hochwimmer <notifications@github.com>:

**Comment by Twonkk** [comment]
it really boggles my mind that no one can get the steps i posted to work, except me... and ive been doing it on 10+ machines now....... /shrug

**Comment by UltimateByte** [comment]
@Twonkk LGSM staff or somebody that reall wants it fix will eventually get it and add the workaround to LGSM.

**Comment by mathieuduperre** [comment]
I confirm this is indeed a problem (and its probably not related to lsgm). We have the same issue on ubuntu 16, and ubuntu 14 (both baremetal and aws).  We tried CS16, cscz, csgo. all the same problem. I even get this error manually with steamcmd (not through lsgm).  Twonkk fix did not work (tried many times in different ways). 

here's a debug log in case someone wants it. i guess i'll look for a tar of the game files for now. 


csczserver@ipx-x-x-x:~$ ./csczserver debug

Counter-Strike: Condition Zero Debug
=================================

Distro: Ubuntu 16.04.2 LTS
Arch:   x86_64
Kernel: 4.4.0-66-generic
Hostname:       ip-x-x-x-x
tmux:   tmux 2.1
Avg Load:       1.00, 1.00, 1.00
Free Memory:    3.9G
Free Disk:      21G
GLIBC required: 2.3.4
Server IP:      x.x.x.x:27015
Server password:        NOT SET

Start parameters:
./hlds_run -game czero -strictportbind +ip x.x.x.x -port 27015 +clientport 27005 +map de_dust2 -maxplayers 16 -debug

Use for identifying server issues only!
Press CTRL+c to drop out of debug mode.
Warning! If cscz-server is already running it will be stopped.

Continue? [Y/n] Y
[ INFO ] Debug cscz-server: Stopping any running servers
[ INFO ] Stopping cscz-server: LinuxGSM is already stopped
[  OK  ] Debug cscz-server: Starting debug
Enabling debug mode
./hlds_run: 1: ./hlds_run: gdb: not found
Please install gdb first.
goto http://www.gnu.org/software/gdb/
Auto-restarting the server on crash

Console initialized.
Using breakpad crash handler
Setting breakpad minidump AppID = 80
Forcing breakpad minidump interfaces to load
Looking up breakpad interfaces from steamclient
Calling BreakpadMiniDumpSystemInit
FATAL ERROR (shutting down): W_LoadWadFile: couldn't load gfx.wad
email debug.log to linux@valvesoftware.com
Thu Jul 13 14:19:16 UTC 2017: Server restart in 10 seconds
[  OK  ] Debug cscz-server: Closing debug

**Comment by williamknn** [comment]
The fix sent by @Twonkk did work for me, in a Ubuntu Server 16.04 LTS.

**Comment by williamknn** [comment]
I search by the error because my server won't starts and when I ran this workaround, I was able to start my server. But now I cannot  find my server. So, I looked deeper and found the same error again. Here's my logs and details: https://hastebin.com/asujifaful

I'm running the server in an AWS EC2 t2.micro instance (should I upgrade the instance?) and this is the ports I've opened in my security group (inbound rules): 

![inbound_rules](https://user-images.githubusercontent.com/3988151/28237078-71000340-690d-11e7-8443-407307435531.png)

**Comment by williamknn** [comment]
@UltimateByte can you help me with this issue, please?

**Comment by michaelshmitty** [comment]
Same problem here. Tried the fixes (logging in with an actual steam account instead of logging in anonymously), always the same problem. gfx.wad not found.
It looks like the steamcommand just doesn't download all the necessary files to run this.
I ran into this issue without LinuxGSM as well, so is this a Steam / Valve problem? Have they just pulled the content? What a pita to get a simple cs server set up...

**Comment by broonu** [comment]
Told you... the download is broken.
Just download the game in a windows machine them send the files via FTP
over the LinuxGSM installation.

2017-07-25 18:17 GMT-03:00 Michael Smith <notifications@github.com>:

**Comment by xuan1943** [comment]
I also met this ： "csserver: FATAL ERROR (shutting down): W_LoadWadFile: couldn't load gfx.wad"
![image](https://user-images.githubusercontent.com/29734418/29546347-7aa4437a-8726-11e7-84d6-4ebdb2d19722.png)
Related to this?

**Comment by JulienRamel** [comment]
So... the only solution is to wait for a download fix from Steam?

Had the same issue and tried to copy the files from my Mac, but now I get...
`FATAL ERROR (shutting down): Couldn't find 'decals.wad' in "DEFAULTGAME" search path`

**Comment by fjeddy** [comment]
Finally got this working after being determined to find a solution. It has nothing to do with the login as @Twonkk suggests. Instead, it's a rather old bug that seems to have re-appeared.

Noticed that there also has been created an issue specific to the matter #1637 

Solution? Simple, Daniel Gibbs has posted it 5 years ago.

You need the manifest files located at https://github.com/dgibbs64/HLDS-appmanifest, tho, the location of where the manifest files needs to go, is not accurate, they need to go into `serverfiles/steamapps`

Basically, if your trying to install Counter Strike 1.6, use the main files in the repo, that would be

```
appmanifest_10.acf
appmanifest_70.acf
appmanifest_90.acf
```

and place them in `serverfiles/steamapps` and install the server again

If you're trying to install Condition Zero, use the `appmanifest_90.acf` in the czero folder instead of the default `appmanifest_90.acf`

Successfully installed the server without any hiccups using those manifest files after reading https://forums.alliedmods.net/showthread.php?t=297163

**Comment by michaelshmitty** [comment]
@fjeddy many thanks for your effort in getting to the bottom of this. I'll try this out as soon as I can and get back here with results. Cheers!

**Comment by dgibbs64** [comment]
I am reviewing my appmanifest repo to provide the fix for this. Thanks Valve for bringing back a really old bug!

**Comment by dgibbs64** [comment]
I have updated the repo https://github.com/dgibbs64/HLDS-appmanifest
